packages/common/container-definitions/src/loader.ts

```diff
@@ -386,15 +404,15 @@ export interface IContainer extends IEventProvider<IContainerEvents> {
 	 * @param error - If the container is being disposed due to error, this provides details about the error that
 	 * resulted in disposing it.
 	 */
-	dispose(error?: ICriticalContainerError): void;
+	dispose(disconnectReason: DisconnectReason, error?: ICriticalContainerError): void;
 
 	/**
 	 * Closes the container.
 	 *
 	 * @param error - If the container is being closed due to error, this provides details about the error that
 	 * resulted in closing it.
 	 */
-	close(error?: ICriticalContainerError): void;
+	close(disconnectReason: DisconnectReason, error?: ICriticalContainerError): void;
```

Comment:
Does anyone implement `IContainer` outside of FF repo?
We will not be able to change the signature if anything does. You can at least run this change against Loop to check there. Senior engineers maybe be able to tell you if anyone implements it outside. May have to investigate code bases.
If a signature change is okay because no one implements it, then we should make it `@sealed`. You will also need to use overload to ensure caller compatibility.